### PR TITLE
Added OAuth2 client rate limit event type

### PR DESCRIPTION
**Pull request by christinechoy-okta**
## Description:
- **What's changed?** Added client rate limit event type info to rate limit system log events
- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-416975](https://oktainc.atlassian.net/browse/OKTA-416975)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md [hunk 1]
```diff
@@ -36,7 +36,14 @@ This event is fired when the framework is in **Log per client** mode and a speci
 * [`system.client.concurrency_rate_limit.notification`](/docs/reference/api/event-types/?q=system.client.concurrency_rate_limit.notification)<br>
 This event is fired when the framework is turned on in **Log per client** mode and a specific client, IP address, Device token combination makes more than two concurrent requests. However, the end user won't see a rate limit violation. Okta fires only a `notification` System Log event. The System Log contains information about the client ID, IP address, Device identifier, and the actual user if the user already has a valid session.
 
-### Operation rate limits
+### OAuth client rate limit
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md:39]:
```suggestion
### OAuth 2.0 client rate limit
```

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md [hunk 2]
```diff
@@ -36,7 +36,14 @@ This event is fired when the framework is in **Log per client** mode and a speci
...
+
+The following OAuth clent System Log event is fired when requests from a single client ID has consumed the majority of an org's rate limit on the OAuth endpoint:
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md:41]:
```suggestion
The following OAuth 2.0 client System Log event is fired when requests from a single client ID has consumed the majority of an org's rate limit on the OAuth endpoint:
```

Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md:41]: Which "OAuth" endpoint?
=> "OAuth 2.0 endpoints"?
=> `/authorize`, `/login/login.htm`  endpoints?
=> all `/oauth2` endpoints?

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md [hunk 3]
```diff
@@ -36,7 +36,14 @@ This event is fired when the framework is in **Log per client** mode and a speci
...
+
+* [`app.oauth2.client_id_rate_limit_warning`](/docs/reference/api/event-types/?q=app.oauth2.client_id_rate_limit_warning)<br>
+This event type is fired when a single client ID consumes 90% of an org's OAuth2 rate limit. (Note that this threshold is subject to change.) This event contains information about the responsible client ID, which can be used by admins to discover and deactivate a rogue client.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md:44]: OAuth2 => OAuth 2.0 ?

**Review by vanngo-okta — COMMENTED** [review]
Sorry for the nit, can you clarify "OAuth endpoint" (and "OAuth2" references)?

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md (hunk 3 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md:44]: This event type has been around for a while as OAuth2. Maybe @peteryaworsky-okta can answer this?

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md [hunk 4]
```diff
@@ -36,7 +36,14 @@ This event is fired when the framework is in **Log per client** mode and a speci
...
 This event is fired when the framework is turned on in **Log per client** mode and a specific client, IP address, Device token combination makes more than two concurrent requests. However, the end user won't see a rate limit violation. Okta fires only a `notification` System Log event. The System Log contains information about the client ID, IP address, Device identifier, and the actual user if the user already has a valid session.
 
-### Operation rate limits
+### OAuth <span class="x x-first x-last">2.0 </span>client rate limit
+
+The following OAuth 2.0 client System Log event is fired when requests from a single client ID has consumed the majority of an org's rate limit on the OAuth endpoint:
+
+* [`app.oauth2.client_id_rate_limit_warning`](/docs/reference/api/event-types/?q=app.oauth2.client_id_rate_limit_warning)<br>
+This event type is fired when a single client ID consumes 90% of an org's OAuth2 rate limit. (Note that this threshold is subject to change.) This event contains information about the responsible client ID, which can be used by admins to discover and deactivate a rogue client.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md:44]:
```suggestion
This event type is fired when a single client ID consumes 90% of an org's OAuth 2.0 rate limit. (Note that this threshold is subject to change.) This event contains information about the responsible client ID, which can be used by admins to discover and deactivate a rogue client.
```

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md (hunk 3 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md]: Fixed

**Review by joec-okta — COMMENTED** [review]
looks good

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md [hunk 5]
```diff
@@ -36,7 +36,14 @@ This event is fired when the framework is in **Log per client** mode and a speci
 * [`system.client.concurrency_rate_limit.notification`](/docs/reference/api/event-types/?q=system.client.concurrency_rate_limit.notification)<br>
 This event is fired when the framework is turned on in **Log per client** mode and a specific client, IP address, Device token combination makes more than two concurrent requests. However, the end user won't see a rate limit violation. Okta fires only a `notification` System Log event. The System Log contains information about the client ID, IP address, Device identifier, and the actual user if the user already has a valid session.
 
-### Operation rate limits
+### OAuth 2.0 client rate limit
+
+The following OAuth 2.0 client System Log event is fired when requests from a single client ID has consumed the majority of an org's rate limit on the `/oauth2` endpoint:
```
Comment (peteryaworsky-okta) [inline packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md:41]: I'd change this to:
`The following System Log event is fired when OAuth 2.0 requests from a single client ID have consumed the majority of the applicable rate limit for the org:`

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md [hunk 6]
```diff
@@ -36,7 +36,14 @@ This event is fired when the framework is in **Log per client** mode and a speci
...
+
+* [`app.oauth2.client_id_rate_limit_warning`](/docs/reference/api/event-types/?q=app.oauth2.client_id_rate_limit_warning)<br>
+This event type is fired when a single client ID consumes 90% of an org's OAuth 2.0 rate limit. (Note that this threshold is subject to change.) This event contains information about the responsible client ID, which can be used by admins to discover and deactivate a rogue client.
```
Comment (peteryaworsky-okta) [inline packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md:44]: "This event type is fired when a single client ID consumes 90% of an org's OAuth 2.0 rate limit" ->
"This event type is fired when OAuth 2.0 requests from a  single client ID consume 90% of the applicable rate limit for the org"

This sounds almost exactly like what we have right above though – do we need the same statement in two places?

I'd also change the "and deactivate a rogue client" part to be something like "and deactivate or reconfigure the client if necessary"

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md (hunk 5 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md:41]: Fixed.

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md [hunk 7]
```diff
@@ -49,11 +56,11 @@ This event type may be sent once per rate limit period as a warning that some si
 * `system.operation.rate_limit.notification`<br>
 This event type can provide additional information about rate limit decisions. For example, this event might indicate that a violation event would have been emitted for a specific client rather than for a broader scope if you had chosen a different configuration.
 
-#### DebugContext object for operation rate limits
+#### DebugContext object for operation rate limit
```
[the remark this answers is not in the document]
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md:59]: Fixed.

File: packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md (hunk 6 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/reference/rl-system-log-events/index.md]: Fixed

**Review by peteryaworsky-okta — APPROVED** [review]

**Review by joec-okta — COMMENTED** [review]
LGTM

**Review by joec-okta — APPROVED** [review]

**Pull request merged**